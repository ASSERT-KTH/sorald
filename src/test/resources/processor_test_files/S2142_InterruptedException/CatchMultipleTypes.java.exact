        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }