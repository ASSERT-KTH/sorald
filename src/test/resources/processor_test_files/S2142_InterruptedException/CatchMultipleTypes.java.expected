/*
We should only set the interrupted flag if an interrupt actually occurred, so when a catch has
multiple types in the catch clause, we need to do a type check.
 */
import java.util.concurrent.ExecutionException;

public class CatchMultipleTypes {
    public static void method() {
        try {
            if (1 < 2) {
                throw new ExecutionException(new RuntimeException());
            } else {
                throw new InterruptedException();
            }
        } catch (ExecutionException e) {
            throw new RuntimeException(e);
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
        }
    }
}